File: .3rd_party/github.com/jackgene/elm-syntax-highlight/demo/examples/syntax.py
from enum import Enum
from typing import *

# Comment
PI: float = 3.14
num: Optional[int] = None
nums: list[int] = [1, 2, 3]
flag: bool = True
iceCream: str = "chocolate" if flag else "vanilla"

if iceCream == 'chocolate':
    print(f'Yay, I love ${iceCream} ice cream!')
else:
    print('Awwww, but chocolate is my favorite...')


def multiply(num1: float, num2: float) -> float:
    result = num1 * num2 * 1.0
    return result


product: float = multiply(num, 10)


class Thing:
    pass


class Polygon(Thing):
    _name: str
    _height: float
    _width: float

    def __init__(self, height: float, width: float):
        super()
        self._name = 'Polygon'
        self._height = height
        self._width = width

    @staticmethod
    def get_name(self) -> str:
        print("hello")
        x = 1 * 2
        return self._name


def make_polygon(height: float) -> Polygon:
    return Polygon(height, product)


p: Polygon = make_polygon(1.4)
p.get_name()


class Direction(Enum):
    NORTH = "n"
    SOUTH = "s"
    EAST = "e"
    WEST = "w"


print(Direction.NORTH)

File: .3rd_party/github.com/jackgene/elm-syntax-highlight/demo/examples/syntax.kt
package syntax

// Some values
val Pi: Any = 3.14 as Any // TODO - make values consistent - start with Go
val chr: Char = 'x'
var byte: Byte? = null
var num: Double = 42.0
val nums: IntArray = intArrayOf(1, 2, 3)
val flag: Boolean = Pi !is Double || false
val iceCream: String = if (flag) "chocolate" else "vanilla"

// Code snippet
if (iceCream == "chocolate") {
    print("Yay, I love \(iceCream) ice cream!")
} else {
    print("Awwww, but chocolate is my favorite...")
}

// Function declaration
fun multiply(num1: Double, num2: Double): Double {
    val result = num1 * num2 * 1.0
    return result
}

// Function evaluation
val product: Double = multiply(num, 10.0)

// Type definition
interface Shape {
    var size: Double
    fun fits(within: Shape): Boolean
}
internal open class Thing(val name: String) {
    // TODO

}
@Deprecated("Don't use me")
internal class Polygon(
        private val height: Double, private val width: Double) :
        Thing("Polygon") {
    init {
        print("hello")
    }

    public override fun getName(): String {
        val _ = 1 * 2
        return super.name
    }

    var area: Double {
        get {
            return multiply(this.height, this.width)
        }
    }
}
extension Polygon: Shape {
    var size: Double {
        get {
            return this.area
        }
    }

    public fun fits(within: Shape): Bool {
        this.size < within.size
    }
}

// Function using declared type
fun makePolygon(height: Double): Polygon {
    return Polygon(height: height, width: product)
}
val s: Shape = makePolygon(height: 1.4)
print(s is Polygon)
guard val p = s as? Polygon else {
    exit(1)
}
print("\(p.getName()) \(p.height + p.width)")

// Enum declaration
enum class Direction {
    NORTH,
    SOUTH,
    EAST,
    WEST
}

fun multiply(num1: Int, num2: Int): Int {
    return num1 * num2
}

@Deprecated("No!")
fun main() {
    val char = 'x'
    print(multiply(42, 24))
}
